_posts/en/newsletters/2020-01-29-newsletter.md

```diff
@@ -0,0 +1,215 @@
+endcomment %}
+{% assign bse = "https://bitcoin.stackexchange.com/a/" %}
+
+- [How is OP_CHECKTEMPLATEVERIFY a scaling solution?]({{bse}}92755)
+  Confused_Coder explains ways in which the proposed new opcode, `OP_CHECKTEMPLATEVERIFY`,
+  could help with more [efficient use of block space over time][news48 output commitments]
+  by encapsulating multiple payments into a single output which are then later claimed onchain.
+
+- [Why was the BIP32 fingerprint used for BIP174 psbt?]({{bse}}92848)
+  Andrew Chow describes why the BIP32 fingerprint was chosen for use throughout
+  the BIP174 [PSBT][topic psbt] specification instead of the full hash, namely
+  the impracticality of getting the full pubkey hash160 from hardware wallets.
+
+- [How is the size of a bitcoin transaction calculated?]({{bse}}92689)
+  User Septem151 provides a detailed outline, broken out by each field, of how
+  the weight units (vbytes) are calculated for both segwit and non-segwit transactions.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[libsecp256k1][libsecp256k1 repo], [Bitcoin Improvement Proposals
+(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #17492][] causes Bitcoin Core GUI to place a
+  Partially-Signed Bitcoin Transaction ([PSBT][topic psbt]) in the
+  clipboard if the user attempts to fee bump a transaction in a
+  watch-only wallet.  The user can then paste the PSBT into another
+  program (such as [HWI][topic hwi]) for signing.
+
+- [C-Lightning #3376][] FIXME:moneyball
+
+- [LND #3809][] adds a `force` parameter to the `BumpFee` RPC so that
+  it can include uneconomical UTXOs in the transactions it creates,
+  extending the change described in [Newsletter #79][news79 lnd3814].
+  Uneconomical UTXOs are UTXOs that cost more to spend than they contain
+  in value---it's important that LND be able to spend these if the
+  proposed [anchor outputs][news70 anchor] fee bumping method is adopted
+  into the LN protocol.
+
+- [BIPs #875][] assigns [BIP119][] to the `OP_CHECKTEMPLATEVERIFY`
+  proposal.  If the proposal is adopted, users will be able to create
+  UTXOs that can only be spent by a specific transaction or set of
+  transactions, providing a type of [covenant][topic covenants].  This is
+  useful in protocols that temporarily keep transactions offchain but
+  which need to provide strong guarantees that, when the transaction
+  goes onchain, it will be exactly the transaction the receiver
+  expected.
+
+- [BIPs #876][] assigns BIPs to the elements of the schnorr-taproot
+  proposal:
+
+    - [BIP340][] is assigned to "Schnorr Signatures for secp256k1,"
+      which describes a signature scheme compatible with the secp256k1
+      [elliptic curve][] used by Bitcoin.  The signatures are
+      compatible with batch verification and with key and signature
```

Comment:
I don't understand.  BIP340 keys and signatures should currently be compatible with batch verification and MuSig.

Reply:
Yes, they are compatible, my thought was to perhaps describe their implementation state.

According to BIP 140, batch signature validation "...code is already in bitcoin/secp256k1 and can be merged in".

That leaves MuSig, which afaik does not have a proposed BIP implementation yet; it is mentioned in the BIPs only in reference to https://eprint.iacr.org/2018/068.

Reply:
Yes, they are compatible, my thought was to perhaps describe their implementation state.

According to BIP 140, batch signature validation "...code is already in bitcoin/secp256k1 and can be merged in".

That leaves MuSig, which afaik does not have a proposed BIP implementation yet; it is mentioned in the BIPs only in reference to https://eprint.iacr.org/2018/068 (and only in BIPs 340-3414-342).

Reply:
and Schnorr key aggregation in https://eprint.iacr.org/2018/068. Happy to be wrong or have misunderstood.

Reply:
Oh, hmm.  I don't think I want to add details about the implementation state to this section about the BIPs.  I think it's enough to know that the scheme proposed in BIP340 is compatible with MuSig (and perhaps other aggregation schemes).  The scheme chosen by users isn't a consensus detail, so it doesn't need a BIP (though it can have a BIP for standardization).  It's like the difference between using `2 <pubkey> <pubkey> 2 OP_CHECKMULTISG` and `<pubkey> OP_CHECKSIGVERIFY <pubkey> OP_CHECKSIG` --- different scripts with different tradeoffs that accomplish the same goal.

Reply:
Agree with Dave that this shouldn't be about implementation. This section is about the BIPs, which describe the standard.

Reply:
Thanks -- I appreciate being able to exchange on these things.